ludwig/encoders/date_encoders.py

```diff
@@ -232,34 +253,33 @@ def __init__(
             default_weights_regularizer=weights_regularizer,
             default_bias_regularizer=bias_regularizer,
             default_activity_regularizer=activity_regularizer,
-            # default_weights_constraint=weights_constraint,
-            # default_bias_constraint=bias_constraint,
             default_norm=norm,
             default_norm_params=norm_params,
             default_activation=activation,
             default_dropout=dropout,
         )
 
-    def call(
+    def forward(
             self,
-            inputs,
-            training=None,
-            mask=None
-    ):
+            inputs: torch.Tensor,
+            training: Optional[bool] = None,
+            mask: Optional[bool] = None
+    ) -> Dict[str, torch.Tensor]:
         """
-            :param input_vector: The input vector fed into the encoder.
-                   Shape: [batch x 19], type tf.int8
-            :type input_vector: Tensor
-            :param training: bool specifying if in training mode (important for dropout)
+            :param inputs: The input vector fed into the encoder.
+                Shape: [batch x DATE_INPUT_SIZE], type torch.int8
+            :type inputs: Tensor
+            :param training: bool specifying if in training mode (important for
+                dropout).
             :type training: bool
-            :param mask: bool specifying masked values
+            :param mask: bool specifying masked values.
             :type mask: bool
          """
         # ================ Embeddings ================
-        input_vector = tf.cast(inputs, tf.int32)
+        input_vector = inputs.type(torch.int32)
```

Comment:
Looks like you cast every input to `torch.float32` before passing it to each embedding module, so you could potentially cast all inputs to `torch.int32` here and remove the casting in line 282 e.g.

Reply:
I believe this may be resolved with the latest push. Please reopen if not.